mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -185,6 +186,56 @@ void AbstractControllerExecution::setVelocityCmd(const geometry_msgs::TwistStamp
   // TODO so there should be no loss of information in the feedback stream
 }
 
+bool AbstractControllerExecution::checkCmdVelIgnored(const geometry_msgs::Twist& cmd_vel)
+{
+  // check if the velocity ignored check is enabled or not
+  if (cmd_vel_ignored_tolerance_ <= 0.0)
+  { 
+    return false;
+  }
+
+  const bool robot_stopped = robot_info_.isRobotStopped(1e-3, 1e-3);
+
+  // compute linear and angular velocity magnitude
+  const double cmd_linear = std::hypot(cmd_vel.linear.x, cmd_vel.linear.y);
+  const double cmd_angular = std::abs(cmd_vel.angular.z);
+
+  const bool cmd_is_zero = cmd_linear < 1e-3 && cmd_angular < 1e-3;
+
+  if (!robot_stopped || cmd_is_zero)
+  {
+    // velocity is not being ignored
+    if (!first_ignored_time_.is_zero())
+    {
+      first_ignored_time_ = ros::Time();
+    }
+    return false;
+  }
+
+  if (first_ignored_time_.is_zero())
+  {
+    // set first_ignored_time_ to now if it was zero
+    first_ignored_time_ = ros::Time::now();
+  }
```

Comment:
```suggestion
```

F: Remove since we will never enter this if statement (`first_ignored_time_` is set in `AbstractControllerExecution::run` before calling this function)

Reply:
This logic is setting the first time robot ignored the cmd_vel. It will be used to calculate ignored_duration when the robot keeps ignoring cmd_vel.